### PR TITLE
proxy: Remove TLS client config from Process context

### PR DESCRIPTION
**Pull request by hawkw**
As the TLS client config watch stored in `ctx::Process` is used only in
`Bind`, it's not necessary for it to be part of the process context.
Instead, it can be explicitly passed into `Bind`.

The resultant code is simpler, and resolves a potential cyclic 
dependency caused when adding `Sensors` to the watch (see 
https://github.com/runconduit/conduit/pull/1141#issuecomment-400082357).

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

File: proxy/src/telemetry/sensor/tls.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pub struct AcceptHandshake {
+    handle: super::Handle,
+}
```
Comment (briansmith) [inline proxy/src/telemetry/sensor/tls.rs]: I think this file doesn't belong in this PR?

Comment (hawkw) [reply proxy/src/telemetry/sensor/tls.rs]: oops, thanks for pointing that out!
